### PR TITLE
txn: Keep pessimistic lock's TTL if it's greater than prewrite request's ttl

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

###  What have you changed?

According to @tiancaiamao 's suggestion, if a pessimistic prewrite request arrives and it carries a smaller TTL than that already wrote in the pessimistic lock, keep the pessimistic lock's TTL rather than use the TTL in the reqeust.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by MyonKeminta** [comment]
@tiancaiamao PTAL

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -362,6 +379,7 @@ impl<S: Snapshot> MvccTxn<S> {
                     return Ok(());
                 }
                 // The lock is pessimistic and owned by this txn, go through to overwrite it.
+                last_lock_ttl = lock.ttl;
```
Comment (coocood) [inline src/storage/mvcc/txn.rs]: Can we update the TTL in the `options` instead of adding an argument?

Comment (coocood) [reply src/storage/mvcc/txn.rs]: Oh, then all other keys will use the greater TTL even if it is not the primary key.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**